hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java

```diff
@@ -244,6 +244,10 @@ public void finalizeBlock(Container container, BlockID blockId)
         db.getStore().getFinalizeBlocksTable().putWithBatch(batch,
             kvContainer.getContainerData().getBlockKey(localID), localID);
         db.getStore().getBatchHandler().commitBatchOperation(batch);
+
+        BlockData emptyBlockData = new BlockData(blockId);
+        db.getStore().putBlockByID(batch, incrementalEnabled, localID,
+            emptyBlockData, kvContainer.getContainerData(), true);
```

Comment:
When `incrementalEnabled` is `true` then `emptyBlockData` should include metadata `INCREMENTAL_CHUNK_LIST` as well so that it should go in [AppendLastChunk](https://github.com/apache/ozone/blob/30978110a63cf302074782d9832142ce44a8f273/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/DatanodeStoreWithIncrementalChunkList.java#L148) to append last chunk into `blockTable.`